internal/local/structure.go

```diff
@@ -230,3 +232,118 @@ func parseServices(rootDir string) ([]ServiceStructure, error) {
 
 	return out, nil
 }
+
+func writeSecrets(rootDir string, secrets *SecretsStructure) error {
+	if secrets == nil {
+		return nil
+	}
+	data, err := MarshalJSON(secrets)
+	if err != nil {
+		return err
+	}
+	return WriteFile(filepath.Join(rootDir, FileSecrets.String()), 0666, bytes.NewReader(data))
+}
+
+func writeEnvironments(rootDir string, environments map[string]map[string]interface{}) error {
+	for env, values := range environments {
+		data, err := MarshalJSON(values)
+		if err != nil {
+			return err
+		}
+		err = WriteFile(filepath.Join(rootDir, NameEnvironments, env), 0666, bytes.NewReader(data))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func writeValues(rootDir string, values []map[string]interface{}) error {
+	if err := os.MkdirAll(filepath.Join(rootDir, NameValues), os.ModePerm); err != nil {
+		return err
+	}
+	for _, vals := range values {
+		data, err := MarshalJSON(vals)
+		if err != nil {
+			return err
+		}
+		err = WriteFile(filepath.Join(rootDir, NameValues, fmt.Sprintf("%s%s", vals["name"], extJSON)), 0666, bytes.NewReader(data))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func writeGraphQL(rootDir string, graphql GraphQLStructure) error {
+	if err := os.MkdirAll(filepath.Join(rootDir, NameGraphQL, NameCustomResolvers), os.ModePerm); err != nil {
+		return err
+	}
+	if graphql.Config != nil {
+		data, err := MarshalJSON(graphql.Config)
+		if err != nil {
+			return err
+		}
+		err = WriteFile(filepath.Join(rootDir, NameGraphQL, FileConfig.String()), 0666, bytes.NewReader(data))
+		if err != nil {
+			return err
+		}
+	}
+	for _, vals := range graphql.CustomResolvers {
+		data, err := MarshalJSON(vals)
+		if err != nil {
+			return err
+		}
+		nameFile := fmt.Sprintf("%s_%s%s", strings.ToLower(vals["on_type"].(string)), strings.ToLower(vals["field_name"].(string)), extJSON)
```

Comment:
we should avoid the possibility of panics with unchecked `.(string)` casts here (even though our data model suggests that'll never happen) by wrapping the whole `fmt.Sprintf` with `strings.ToLower`

also, its hard for me not to just suggest naming this `filename`, _crazy with go-style names we dont have to be_

---

sidenote, this is another good example of the benefits to having a shared library holding these import/export app structure definitions